pkg/controller/elasticsearch/certificates/transport/reconcile.go

```diff
@@ -39,8 +40,24 @@ func ReconcileTransportCertificatesSecrets(
 	rotationParams certificates.RotationParams,
 ) *reconciler.Results {
 	results := &reconciler.Results{}
+
+	// We must create transport certificates for the following StatefulSets:
+	// - the ones that still exist, even if they have been removed from the Spec
+	// - the ones that do not exist yet, but will be created in a later step of the reconciliation
+	actualStatefulSets, err := sset.RetrieveActualStatefulSets(c, k8s.ExtractNamespacedName(&es))
+	if err != nil {
+		return results.WithError(err)
+	}
+	ssets := make(map[string]struct{})
+	for _, actualStatefulSet := range actualStatefulSets {
+		ssets[actualStatefulSet.Name] = struct{}{}
+	}
 	for _, nodeSet := range es.Spec.NodeSets {
-		if err := reconcileNodeSetTransportCertificatesSecrets(c, ca, es, nodeSet.Name, rotationParams); err != nil {
+		ssets[esv1.StatefulSet(es.Name, nodeSet.Name)] = struct{}{}
```

Comment:
The one thing I wonder is whether we could go one step further an anticipate the Pods that are about to be created and generate their certificates ahead of time. Because if I understand the flow correctly currently we will just create a secret with the CA cert in it and it takes at least another reconciliation iteration to then provision the certificates for the Pods which are hanging in the init step until the certs appear. 

This would be a bit of a bigger refactoring and if we think it is useful, it would be OK imo to do it in a separate PR and now just fix the broken behaviour.

Reply:
We need their IP, I think, so we need to wait for the Pods to be created anyway.

Reply:
Oh you are right we reverted that change that used DNS names in the `network.publish_host`. I forgot about that.